module/VuFind/src/VuFind/RecordDriver/ProQuestFSG.php

```diff
@@ -0,0 +1,129 @@
+ * @link     https://vufind.org/wiki/development:plugins:record_drivers Wiki
+ */
+
+namespace VuFind\RecordDriver;
+
+use VuFind\RecordDriver\Feature\MarcAdvancedTrait;
+use VuFind\RecordDriver\Feature\MarcBasicTrait;
+use VuFind\RecordDriver\Feature\MarcReaderTrait;
+
+/**
+ * Model for MARC records in ProQuest Federated Search Gateway.
+ *
+ * @category VuFind
+ * @package  RecordDrivers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_drivers Wiki
+ */
+class ProQuestFSG extends DefaultRecord
+{
+    use MarcReaderTrait, MarcAdvancedTrait, MarcBasicTrait {
+        MarcBasicTrait::getNewerTitles insteadof MarcAdvancedTrait;
+        MarcBasicTrait::getPreviousTitles insteadof MarcAdvancedTrait;
+        MarcBasicTrait::getShortTitle as marcGetShortTitle;
+        MarcBasicTrait::getTitle as marcGetTitle;
+    }
+
+    /**
+     * Set raw data to initialize the object.
+     *
+     * @param mixed $data Raw data representing the record; Record Model
+     * objects are normally constructed by Record Driver objects using data
+     * passed in from a Search Results object. In this case, $data is a MARCXML
+     * document.
+     *
+     * @return void
+     */
+    public function setRawData($data)
+    {
+        // Ensure that $driver->setRawData($driver->getRawData()) doesn't blow up:
+        if (isset($data['fullrecord'])) {
+            $data = $data['fullrecord'];
+        }
+
+        // Map the ProQuestFSG response into a format that the parent Solr-based
+        // record driver can understand.
+        parent::setRawData(['fullrecord' => $data]);
+    }
+
+    /**
+     * Get the short (pre-subtitle) title of the record.
+     *
+     * @return string
+     */
+    public function getShortTitle()
+    {
+        return strip_tags($this->marcGetShortTitle());
+    }
+
+    /**
+     * Get the full title of the record.
+     *
+     * @return string
+     */
+    public function getTitle()
+    {
+        return strip_tags($this->marcGetTitle());
+    }
+
+    /**
+     * Get the title of the item that contains this record (i.e. MARC 773s of a
+     * journal).
+     *
+     * @return string
+     */
+    public function getContainerTitle()
+    {
+        return $this->getFirstFieldValue('773', ['t']);
+    }
+
+    /**
+     * Get a full, free-form reference to the context of the item that contains this
+     * record (i.e. volume, year, issue, pages).
+     *
+     * @return string
+     */
+    public function getContainerReference()
+    {
+        return $this->getFirstFieldValue('773', ['g']);
+    }
+
+    /**
+     * Get the item's source.
+     *
+     * @return string
+     */
+    public function getSource()
+    {
+        return $this->getFirstFieldValue('786', ['t']);
+    }
```

Comment:
I'm not clear what kind of data `source` means here as it's only used in Primo and I don't have that.  I'm assuming it's the name of the database.

Reply:
Never mind, answered my own question, yes it's the database.